src/nft/components/bag/BagFooter.tsx

```diff
@@ -316,6 +317,7 @@ export const BagFooter = ({ setModalIsOpen, eventProperties }: BagFooterProps) =
   const isPending = PENDING_BAG_STATUSES.includes(bagStatus)
   const activeCurrency = inputCurrency ?? defaultCurrency
   const usingPayWithAnyToken = !!inputCurrency && chainId === SupportedChainId.MAINNET
+  const universalRouterAddress = useNftUniversalRouterAddress()
```

Comment:
`useNftUniversalRouterAddressQuery` also as an optional `loading` field you could pass through this hook as `universalRouterAddress` would be undefined while loading. While`universalRouterAddressIsLoading` is true you should set `tradeState` to Loading to prevent an inaccurate UR address from being used because the query was still loading.